documentation/library_guide/parallel_api/pass_data_algorithms.rst

```diff
@@ -81,7 +81,19 @@ the buffer were created for the same queue. For example:
     return 0;
   }
 
-Alternatively, use ``std::vector`` with a USM allocator. For example:
+Alternatively, use ``std::vector`` with a USM allocator.
+Note: The ability to appropriately detect USM allocated ``std::vector::iterator`` depends
+on details of the C++ standard library implementation and what information about the
+allocator is included in the ``std::vector::iterator`` type definition. You can check
+if your C++ standard library implementation provides enough information by checking the
+value of ``oneapi::dpl::usm_allocated_vector_iterators_supported_v<ValueType>``. This will
+evaluate to ``true`` for C++ standard library implementations which allow oneDPL to support
+USM allocators with this ValueType and ``false`` otherwise. If usm_allocators are not
+supported with your C++ standard library, they will still function as inputs to oneDPL, but
+they will be treated as if they were host-side ``std::vector::iterator`` as described in
+the section below.
```

Comment:
Its worth discussing if it is a good idea to expose this information to the end user, and if this is the best way to convey it.  

I don't think its possible to do better without forcing the user to mark the input data via some wrapper (which doesn't seem better or easier than using `std::vector::data()` to get a usm pointer instead.

Reply:
Another necessary bit of information:  If we cannot detect USM allocated data, it is treated like a host iterator.  Host iterators are not supported as the source iterator for `permutation_iterator`, as it cannot be efficiently determined what range of data should be copied into a `sycl::buffer` for use on the device.  
Therefore, based on the C++ standard library implementation, some uses case may succeed or not succeed.  It seems problematic to not provide some programmatic way to detect which scenario the user is in (`usm_allocated_vector_iterators_supported_v`).  The other option would be to disallow any `std::vector::iterator` as source iterator for `permutation_iterator` with a device policy (rather than just "host-side iterator" as is stated in the docs currently).